pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -192,67 +161,257 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
 }
 
 /**
- * Windows only. Read settings values from registry using schemaObj as a template.
- * @param schemaObj the object used as a template for navigating registry.
- * @param regKey the registry key obtained from nativeReg.openKey().
- * @returns null, or the registry data as an object.
+ * Win32DeploymentReader - encapsulate details about the registry in this class.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
-  let newObject: RecursivePartial<settings.Settings>|null = null;
+class Win32DeploymentReader {
+  protected registryPathProfile: string[];
+  protected keyName = '';
+
+  constructor(registryPathProfile: string[]) {
+    this.registryPathProfile = registryPathProfile;
+  }
 
-  const schemaKeys = Object.keys(schemaObj);
-  // ignore case
-  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
-  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
+  readProfile(): settings.DeploymentProfileType {
+    const [DEFAULTS_HIVE_NAME, LOCKED_HIVE_NAME] = ['Defaults', 'Locked'];
+    let defaults: RecursivePartial<settings.Settings> = {};
+    let locked: RecursivePartial<settings.Settings> = {};
 
-  for (const k of commonKeys) {
-    const schemaVal = schemaObj[k];
-    let regValue: any = null;
+    for (const keyName of ['HKLM', 'HKCU'] as const) {
+      this.keyName = keyName;
+      const key = nativeReg[keyName];
+      const registryKey = nativeReg.openKey(key, this.registryPathProfile.join('\\'), nativeReg.Access.READ);
```

Comment:
This would have to be tested (later, when I have access to a windows machine again). But right now I'm making both value-queries (access code 0x0001) and enumerating sub-keys (access code 0x0008), and `Access.READ == 0x2_0019` while `Access.QUERY_VALUE` is just `0x0001` so I don't think we can enumerate the sub-keys with that access.

To check later.